### PR TITLE
RRD: Add status details to RRD notification

### PR DESCRIPTION
**Pull request by yoomlam**
## Description of change

Add status details to RRD internal notification, including:
- contention label (e.g., "hypertension" or "asthma")
- RRD PDF created?
- RRD PDF uploaded to S3?
- RRD PDF to be uploaded?
- RRD special issue set?

## Original issue(s)
Prior notification lacked enough detail about the status of the claim submission.

## Things to know about this PR

Only changes internal notification message.
This PR includes PR #9640, so we can wait until that is merged.

### CODE REVIEW

File: spec/services/rapid_ready_for_decision/fast_track_pdf_upload_manager_spec.rb [hunk 1]
```diff
@@ -69,19 +69,15 @@
     end
   end
 
-  describe '#already_has_summary_files' do
+  describe 'rrd_pdf_added_for_uploading?' do
```
Comment (yoomlam) [inline spec/services/rapid_ready_for_decision/fast_track_pdf_upload_manager_spec.rb:72]: Use new method in RRD form526_submission concern instead

File: spec/jobs/rapid_ready_for_decision/rrd_processor_spec.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@
     end
   end
 
-  describe '#set_special_issue', :vcr do
+  describe '#set_special_issue' do
```
Comment (yoomlam) [inline spec/jobs/rapid_ready_for_decision/rrd_processor_spec.rb:23]: Don't need VCR here.

File: app/mailers/rrd_completed_mailer.rb [hunk 3]
```diff
@@ -3,8 +3,8 @@
 class RrdCompletedMailer < ApplicationMailer
   def build(submission)
     @submission = submission
-    @disability = submission.form.dig('form526', 'form526', 'disabilities')&.first
-    @disability_code = @disability['diagnosticCode'] if @disability
+    @disability_struct = RapidReadyForDecision::Constants.first_disability(submission) || {}
+    @disability_code = @disability_struct['code']
```
Comment (yoomlam) [inline app/mailers/rrd_completed_mailer.rb:7]: Use `disability_struct` so we can use the `disability_struct[:label]` in the erb file.

File: app/services/rapid_ready_for_decision/rrd_special_issue_manager.rb [hunk 4]
```diff
@@ -12,7 +12,7 @@ def add_special_issue
       submission_data = JSON.parse(submission.form_json)
       disabilities = submission_data.dig('form526', 'form526', 'disabilities')
       disabilities.each do |disability|
-        add_rrd_code(disability) if hypertension_increase?(disability)
+        add_rrd_code(disability) if included_disability_increase?(disability)
```
Comment (yoomlam) [inline app/services/rapid_ready_for_decision/rrd_special_issue_manager.rb:15]: Changes to this file will not show once #9640 is merged.

File: spec/services/rapid_ready_for_decision/rrd_special_issue_manager_spec.rb [hunk 5]
```diff
@@ -12,7 +12,7 @@ def form526_hash(form)
 
   def filter_disabilities(form)
     form[:disabilities].filter do |item|
-      item[:diagnosticCode] == RapidReadyForDecision::Constants::DISABILITIES[:hypertension][:code]
+      RapidReadyForDecision::Constants::DISABILITIES_BY_CODE.include?(item[:diagnosticCode])
```
Comment (yoomlam) [inline spec/services/rapid_ready_for_decision/rrd_special_issue_manager_spec.rb:15]: Changes to this file will not show once #9640 is merged.

**Review by leikkisa — DISMISSED** [review]
Added a couple of non-blocking comments (this PR still looks great without addressing them) - looks like we need to update the document title!

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb [hunk 6]
```diff
@@ -45,6 +44,28 @@ def pending_eps?
...
+  end
+
+  def rrd_pdf_uploaded_to_s3?
+    form.dig('rrd_metadata', 'pdf_guid').present?
+  end
+
+  # @return if an RRD pdf has been included as a file to upload
+  def rrd_pdf_added_for_uploading?
+    form['form526_uploads']&.any? do |upload|
+      upload['name']&.start_with? RapidReadyForDecision::FastTrackPdfUploadManager::DOCUMENT_TITLE
```
[the remark this answers is not in the document]
Comment (yoomlam) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:58]: I'll address in another PR.

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb [hunk 7]
```diff
@@ -54,6 +75,6 @@ def open_claims
 
   # @return if this claim submission was processed and fast-tracked by RRD
   def rrd_claim_processed?
-    form_json.include? RapidReadyForDecision::FastTrackPdfUploadManager::DOCUMENT_TITLE
+    rrd_pdf_added_for_uploading?
```
[the remark this answers is not in the document]
Comment (yoomlam) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb]: Agree! Updated to `rrd_pdf_added_for_uploading? && rrd_special_issue_set?` to capture the semantics of the method.

**Review by leikkisa — APPROVED** [review]
👍

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (yoomlam) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:58]: Addressed in PR #9702

**Review by RachalCassity — APPROVED** [review]

**Pull request merged**